### PR TITLE
Convert Filebeat elasticsearch.* to ECS

### PR DESCRIPTION
**Pull request by webmat**
Very little appears to be required for this module.

Caveats

- elasticsearch.deprecation, elasticsearch.gc and elasticsearch.server look good already, no need to modify.
- [x] Slowlog has two representations for event duration. One is an int in milliseconds, which is what I used to populate event.duration. But if we parse the textual slowlog.took instead, we can get much more precise durations. Outside of the scope of this ECS mapping, however.

Renames

- elasticsearch.audit.event_type => event.type
- elasticsearch.audit.origin_address => source.ip
- elasticsearch.audit.uri => url.original
- elasticsearch.audit.request_body => http.request.body
- elasticsearch.audit.principal => user.name

Copies

- Populate event.duration (ns) without removing slowlog.took and slowlog.took_millis

TODO

- [x] Generate a few more \`-expected.json\` files prior to making changes
- [x] Save slowlog.took_millis * 1000000 to event.duration
- [x] Make sure to address the `http.*.body.content` move in ecs-migration and the code as well
  - Delete http.request.body from ecs-migration
- [x] Alias renamed fields to their ECS counterpart
- [x] Document field migrations in ecs-migration.yml
- [x] Changelog
- [x] Migrate principal to user.name

### CODE REVIEW

**Comment by ruflin** [comment]
+1 on the Caveats

**Comment by webmat** [comment]
I'll be able to start on this again once #9645 is merged.

File: filebeat/module/elasticsearch/audit/test/test.log-expected.json [hunk 1]
```diff
@@ -1,107 +1,107 @@
...
         "elasticsearch.audit.principal": "i030648", 
-        "elasticsearch.audit.uri": "/_xpack/security/_authenticate", 
         "event.dataset": "audit", 
         "event.module": "elasticsearch", 
+        "event.type": "authentication_failed", 
         "input.type": "log", 
         "log.offset": 0, 
         "message": "[2018-06-19T05:16:15,549] [rest] [authentication_failed]        origin_address=[147.107.128.77], principal=[i030648], uri=[/_xpack/security/_authenticate]", 
-        "service.name": "elasticsearch"
+        "service.name": "elasticsearch",
```
Comment (ruflin) [inline filebeat/module/elasticsearch/audit/test/test.log-expected.json]: For later: These should be renamed to service.type.

**Comment by ycombinator** [comment]
Changes LGTM but other related PRs got merged recently so conflicts will need to be resolved on this one. Let me know if you need help with any of those.

File: filebeat/module/elasticsearch/audit/test/test.log-expected.json (hunk 1 above)
Comment (ycombinator) [reply filebeat/module/elasticsearch/audit/test/test.log-expected.json]: Out of curiosity, why is this? I've been seeing `service.name` everywhere in ECS-ready events so far.

Comment (ruflin) [reply filebeat/module/elasticsearch/audit/test/test.log-expected.json]: The definition of `service.name` and `service.type` has change slightly in ECS: https://github.com/elastic/ecs#service `service.name` could be the cluster name of Elasticsearch for example, `service.type` is the one that never changes.

**Comment by webmat** [comment]
Yeah I want to get more module migrations going and get those discussions going first. I'll fix these conflicts in a few days

**Comment by webmat** [comment]
jenkins, test this

**Comment by webmat** [comment]
jenkins, test this

**Comment by webmat** [comment]
@ruflin @ycombinator Ready for a final review. All elasticsearch-related tests are successful here, even in Jenkins.

The [failing filebeat testsuite](https://beats-ci.elastic.co/job/elastic+beats+pull-request+multijob-linux/3575/beat=filebeat,label=ubuntu/console) in there is actually puzzling to me: all tasks are `[success]`, and it concludes that the test run has failed.

But all of this is obviously unrelated to the ES Filebeat module :-)

Only caveat I'd like confirmation on: right now it would be possible to populate event.duration with nanoseconds precision, but I'd have to parse the textual field `slowlog.took`. I say we can call that an improvement for later. The important part for FF is to move things to the right fields. Do you agree?

**Comment by ruflin** [comment]
jenkins, test this

**Comment by webmat** [comment]
[This last run](https://beats-ci.elastic.co/job/elastic+beats+pull-request+multijob-linux/3582/) only had flakiness for libbeat and metricbeat, so we're good.

**Comment by ruflin** [comment]
@webmat Let's get it in. Super unhappy about the CI failures but filebeat is excluded as you mentioned. 

Can you also remove the `WIP` from the PR title?

**Pull request merged**